tests/0126-oauthbearer_oidc.c

```diff
@@ -33,64 +33,140 @@
+        test_consumer_subscribe(c1, topic);
+        test_consumer_poll("OIDC.C1", c1, testid, 1, -1, 1, NULL);
+
+        test_consumer_close(c1);
+
+        rd_kafka_destroy(p1);
+        rd_kafka_destroy(c1);
+        SUB_TEST_PASS();
+}
+
+
+static void
+auth_error_cb(rd_kafka_t *rk, int err, const char *reason, void *opaque) {
+        if (err == RD_KAFKA_RESP_ERR__AUTHENTICATION ||
+            err == RD_KAFKA_RESP_ERR__ALL_BROKERS_DOWN)
+                TEST_SAY("Expected error: %s: %s\n", rd_kafka_err2str(err),
+                         reason);
+        else
+                TEST_FAIL("Unexpected error: %s: %s", rd_kafka_err2str(err),
+                          reason);
+        rd_kafka_yield(rk);
+}
+
+
+/**
+ * @brief After config OIDC, if the token is not valid, make sure the
+ *        authentication fail as expected.
+ *
+ */
+static void
+do_test_produce_consumer_with_OIDC_should_fail(rd_kafka_conf_t *conf) {
+        if (test_needs_auth()) {
+                TEST_SKIP("Mock cluster does not support SSL/SASL\n");
                 return;
         }
 
-        if (res != RD_KAFKA_CONF_OK)
-                TEST_FAIL("%s", errstr);
+        rd_kafka_t *c1;
+        uint64_t testid;
 
-        test_conf_set(conf, "sasl.oauthbearer.client.id", "randomuniqclientid");
-        test_conf_set(conf, "sasl.oauthbearer.client.secret",
-                      "randomuniqclientsecret");
-        test_conf_set(conf, "sasl.oauthbearer.client.secret",
-                      "randomuniqclientsecret");
-        test_conf_set(conf, "sasl.oauthbearer.extensions",
-                      "supportFeatureX=true");
-        test_conf_set(conf, "sasl.oauthbearer.token.endpoint.url",
-                      "https://localhost:1/token");
+        SUB_TEST("Test OAUTHBEARER/OIDC failing with invalid JWT");
 
-        testid = test_id_generate();
-        rd_kafka_conf_set_dr_msg_cb(conf, test_dr_msg_cb);
-        rk = test_create_handle(RD_KAFKA_PRODUCER, conf);
+        test_conf_set(conf, "sasl.oauthbearer.token.endpoint.url",
+                      "http://localhost:8080/retrieve/badformat");
 
-        topic = test_mk_topic_name("0126-oauthbearer_oidc", 1);
-        test_create_topic(rk, topic, 1, 1);
+        rd_kafka_conf_set_error_cb(conf, auth_error_cb);
 
-        /* Produce messages */
-        test_produce_msgs2(rk, topic, testid, 1, 0, 0, NULL, 0);
+        testid = test_id_generate();
 
-        /* Verify messages were actually produced by consuming them back. */
-        test_consume_msgs_easy(topic, topic, 0, 1, 1, NULL);
+        c1 = test_create_consumer("OIDC.fail.C1", NULL, rd_kafka_conf_dup(conf),
+                                  NULL);
 
-        rd_kafka_destroy(rk);
+        rd_kafka_consumer_poll(c1, 10 * 1000);
 
+        test_consumer_close(c1);
+        rd_kafka_destroy(c1);
         SUB_TEST_PASS();
 }
 
 
 int main_0126_oauthbearer_oidc(int argc, char **argv) {
```

Comment:
Since these tests supposedly requires a cluster configured with SASL OAUTHBEARER OIDC I think you need to:
 * skip all tests here if OIDC is not configured
 * rely on the test conf (from test_conf_init()) to have correct OIDC configurations, since htat is set up by trivup, et.al.